python/paddle/device.py

```diff
@@ -158,19 +166,28 @@ def _convert_to_place(device):
             device_id = device_info_list[1]
             device_id = int(device_id)
             place = core.XPUPlace(device_id)
+        if avaliable_npu_device:
+            if not core.is_compiled_with_npu():
+                raise ValueError(
+                    "The device should not be {}, since PaddlePaddle is "
+                    "not compiled with NPU".format(avaliable_npu_device))
+            device_info_list = device.split(':', 1)
+            device_id = device_info_list[1]
+            device_id = int(device_id)
+            place = core.NPUPlace(device_id)
     return place
 
 
 def set_device(device):
     """
-    Paddle supports running calculations on various types of devices, including CPU, GPU and XPU.
+    Paddle supports running calculations on various types of devices, including CPU, GPU, XPU and NPU.
     They are represented by string identifiers. This function can specify the global device
     which the OP will run.
 
     Parameters:
         device(str): This parameter determines the specific running device.
-            It can be ``cpu``, ``gpu:x`` and ``xpu:x``, where ``x`` is the 
-            index of the GPUs or XPUs. 
+            It can be ``cpu``, ``gpu:x``, ``xpu:x`` and ``npu:x``, where ``x`` is the
```

Comment:
It can be ``cpu``, ``gpu``, ``xpu``, ``gpu:x``, ``xpu:x`` and ``npu:x``

Reply:
done